lib/parse.js

```diff
@@ -114,6 +113,9 @@ module.exports = warner(class Parser extends EE {
     this[ABORTED] = false
     this[SAW_NULL_BLOCK] = false
     this[SAW_EOF] = false
+
+    this.on('end', () => setTimeout(() => this.emit('close')))
```

Comment:
Would love something better that an empty setTimeout here. Maybe someone has some thoughts?

Reply:
You might be able to use `queueMicrotask` or `Promise.resolve().then` to avoid waiting for a next tick of the event loop.

Reply:
Thanks @bergundy ! `queueMicrotask` was what I was looking for, and seems to work great 👍

Reply:
Ahhhh.. good point 👍 ok. I'll make that change. Thanks 😊

Reply:
ok @bergundy switched to nextTick to support node 10.

Reply:
npm already uses features that aren't supported in Node 10. We tend to drop support for Node versions as they go out of LTS.

Reply:
I did notice that node 10 was in the testing versions.

Reply:
Due to the heavy use of this package outside of `npm/cli`, I decided to keep the engines at `>=10` to avoid a major release. So it's good to use `nextTick` for now.

@webark one minor nit: would you be able to add a comment flagging that for updating when node 10 support is dropped? something like `TODO: change to queueMicrotask when node 10 support is dropped`